src/components/AdminForm.js

```diff
@@ -1,38 +1,91 @@
-import React from "react";
+import React, { useState, useEffect } from "react";
 import { connect } from "react-redux";
+import { Redirect } from "react-router-dom";
+
 import {
   setCohortName,
-  setCohortType
+  setCohortType,
+  postFormDetailsThunk,
+  setResetApp,
+  resetIsSubmitted
 } from "../redux/actions/adminFormActions";
+import { resetDates } from "../redux/actions/dateActions";
 
 import TextInput from "./TextInput";
 import Select from "./Select";
-import DatePickerContainer from "./DatePickerContainer";
 import SubmitButton from "./SubmitButton";
-import Questions from './Questions/Questions';
-
+import DatePickerContainer from "./DatePickerContainer";
+import Questions from "./Questions/Questions";
 
 const AdminForm = props => {
+  const [isDuplicate, setDuplicate] = useState(false);
+  console.log(props);
+  useEffect(() => {
+    props.setResetApp();
+    props.resetDates();
+  }, []);
+
   const handleCohortNameChange = e => {
     props.setCohortName(e.target.value);
   };
 
-  const handleCohortTypeChange = e => {
-    props.setCohortType(e.target.value);
+  /**
+   * Check if the cohort name and type already exists in databases
+   * If that already exists then the returned length will be greater than 0
+   */
+  const isCohortDuplicate = () => {
+    const { cohortName, cohortType, existingCohorts } = props;
+    return existingCohorts
+      .filter(cohort => cohort.cohortType === cohortType)
+      .filter(
+        cohort => cohort.cohortName.toLowerCase() === cohortName.toLowerCase()
+      ).length;
   };
 
   const handleSubmit = e => {
     e.preventDefault();
-    alert("Submitting");
-    // on submit we need to save the form in DB
+    const {
+      cohortName,
+      cohortType,
+      dateOpen,
+      dateClose,
+      dateOfResponse
+    } = props;
+
+    const isCohortDuplicateValue = isCohortDuplicate();
```

Comment:
I think we should rename this variable into cohortDuplicateCount or cohortDuplicateValue, taking out the 'is', because when "is" is used in a variable name, the best practice is for that variable to be a boolean. (e.g. let isDone = false; ) So people reading code when they see 'is' or 'are' in the beginning of the variable name assume it's a boolean. Changing the name would solve that issue without having to change names/functionality in other places.